spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/discovery/KubernetesServiceInstance.java

```diff
@@ -16,211 +16,17 @@
-	}
-
-	// Allows for deserialization
-	public KubernetesServiceInstance() {
-	}
-
-	@Override
-	public String getInstanceId() {
-		return this.instanceId;
-	}
-
-	@Override
-	public String getServiceId() {
-		return this.serviceId;
-	}
-
-	@Override
-	public String getHost() {
-		return this.host;
-	}
-
-	@Override
-	public int getPort() {
-		return this.port;
-	}
-
-	@Override
-	public boolean isSecure() {
-		return this.secure;
-	}
 
-	@Override
-	public URI getUri() {
-		return uri;
-	}
-
-	public Map<String, String> getMetadata() {
-		return this.metadata;
-	}
-
-	@Override
-	public String getScheme() {
-		return isSecure() ? HTTPS : HTTP;
-	}
-
-	private URI createUri(String scheme, String host, int port) {
-		return URI.create(scheme + "://" + host + ":" + port);
-	}
-
-	public String getNamespace() {
-		return namespace != null ? namespace : this.metadata.get(NAMESPACE_METADATA_KEY);
-	}
-
-	public String getCluster() {
-		return this.cluster;
-	}
-
-	public void setInstanceId(String instanceId) {
-		this.instanceId = instanceId;
-	}
-
-	public void setServiceId(String serviceId) {
-		this.serviceId = serviceId;
-	}
-
-	public void setHost(String host) {
-		this.host = host;
-	}
-
-	public void setPort(int port) {
-		this.port = port;
-	}
-
-	public void setUri(URI uri) {
-		this.uri = uri;
-	}
-
-	public void setSecure(Boolean secure) {
-		this.secure = secure;
-	}
-
-	public void setMetadata(Map<String, String> metadata) {
-		this.metadata = metadata;
-	}
-
-	public void setNamespace(String namespace) {
-		this.namespace = namespace;
-	}
-
-	public void setCluster(String cluster) {
-		this.cluster = cluster;
-	}
-
-	public Boolean getSecure() {
-		return secure;
-	}
-
-	@Override
-	public boolean equals(Object o) {
-		if (this == o) {
-			return true;
-		}
-		if (o == null || getClass() != o.getClass()) {
-			return false;
-		}
-		KubernetesServiceInstance that = (KubernetesServiceInstance) o;
-		return port == that.port && Objects.equals(instanceId, that.instanceId)
-				&& Objects.equals(serviceId, that.serviceId) && Objects.equals(host, that.host)
-				&& Objects.equals(uri, that.uri) && Objects.equals(secure, that.secure)
-				&& Objects.equals(metadata, that.metadata) && Objects.equals(getNamespace(), that.getNamespace())
-				&& Objects.equals(cluster, that.cluster);
-	}
-
-	@Override
-	public String toString() {
-		ToStringCreator creator = new ToStringCreator(this);
-		creator.append("instanceId", instanceId);
-		creator.append("serviceId", serviceId);
-		creator.append("host", host);
-		creator.append("port", port);
-		creator.append("uri", uri);
-		creator.append("secure", secure);
-		creator.append("namespace", getNamespace());
-		creator.append("cluster", cluster);
-		creator.append("metadata", metadata);
+/**
+ * @author wind57
+ *
+ * {@link ServiceInstance} with additional methods, specific to kubernetes.
+ */
+sealed interface KubernetesServiceInstance extends ServiceInstance permits KubernetesAwareServiceInstance {
```

Comment:
I added this to begin with, since there are two more methods that are k8s specific. So it's the same as `ServiceInstance` + two more methods.